compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -515,6 +516,94 @@ SortOp::reifyResultShapes(OpBuilder &b,
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+namespace {
+
+/// This pattern removes unused results from SortOp. The SortOp uses the
+/// Destination Passing Style interface so it's results are tied to it's
+/// operands as well as it's comparitor block arguments. So, to remove unused
+/// results we must also remove the associated operands and block arguments.
+/// 
+/// This will look like:
+/// ```
+/// %0:2 = iree_linalg_ext.sort dimension(1) outs(%arg0, %arg1: tensor<?x10xf32>,
+/// tensor<?x10xi64>) {
+///   ^bb0(%arg2: f32, %arg3: f32, %arg4: i64, %arg5: i64):
+///    %42 = arith.cmpf oge, %arg2, %arg3 : f32
+///    iree_linalg_ext.yield %42 : i1
+/// } -> tensor<?x10xf32>, tensor<?x10xi64>
+/// ```
+/// ->
+/// %0 = iree_linalg_ext.sort dimension(1) outs(%arg0: tensor<?x10xf32>) {
+///   ^bb0(%arg2: f32, %arg3: f32):
+///    %42 = arith.cmpf oge, %arg2, %arg3 : f32
+///    iree_linalg_ext.yield %42 : i1
+/// } -> tensor<?x10xf32>
+/// ```
+/// Note: that we will not remove unused results if their associated block
+/// arguments are used within the comparitor because that's needed for op
+/// functionality.
+///
+struct RemoveUnusedSortOpResults
+    : public OpRewritePattern<IREE::LinalgExt::SortOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(IREE::LinalgExt::SortOp sortOp,
+                                PatternRewriter &rewriter) const override {
+    // To avoid problems in dispatches associated with unused results, prune
+    // them here.
+    Location loc = sortOp->getLoc();
+    auto operands = sortOp.getOutputs();
+    auto results = sortOp.getResults();
+    unsigned numRes = sortOp.getNumResults();
+
+    // If there are no results, bufferization has happened.
+    // Do not try to prune results anymore.
+    // # TODO(#20831): Implement a way to remove unused results when using buffer
+    // semantics.
+    if (numRes == 0) {
+      return failure();
+    }
```

Comment:
I think my comment was not addressed correctly. This is what I'm looking for:

```suggestion
    // # TODO(#20831): Add support for removing unused operands when the op has pure
    // buffer semantics.
    if (sortOp.hasPureBufferSemantic()) {
      return failure();
    }
```